optuna/visualization/contour.py

```diff
@@ -0,0 +1,205 @@
+import math
+
+from optuna.logging import get_logger
+from optuna.structs import StudyDirection
+from optuna.structs import TrialState
+from optuna.study import Study  # NOQA
+from optuna import type_checking
+from optuna.visualization.utils import _check_plotly_availability
+from optuna.visualization.utils import _is_log_scale
+from optuna.visualization.utils import is_available
+
+if type_checking.TYPE_CHECKING:
+    from typing import List  # NOQA
+    from typing import Optional  # NOQA
+    from typing import Tuple  # NOQA
+
+    from optuna.structs import FrozenTrial  # NOQA
+
+    if is_available():
```

Comment:
I don't think this availability check is necessary because this clause is only used during the type checking, and we can assume `plotly` is usually installed when we run `mypy`. Please remove it if you do not have specific reasons.

Reply:
Thank you very much for the good catch in spite of a bit drastic change.